Jenkinsfile

```diff
@@ -0,0 +1,120 @@
+@Library('github.com/mozmar/jenkins-pipeline@20170206.1')
+def stage_deployed = false
+def config
+def docker_image
+
+conduit {
+  node {
+    stage("Prepare") {
+      checkout scm
+      setGitEnvironmentVariables()
+
+      try {
+        config = readYaml file: "jenkins.yml"
+      }
+      catch (e) {
+        config = []
+      }
+      println "config ==> ${config}"
+
+      if (!config || (config && config.pipeline && config.pipeline.enabled == false)) {
+        println "Pipeline disabled."
+      }
+    }
+
+    docker_image = "${config.project.docker_name}:${GIT_COMMIT_SHORT}"
+
+    stage("Build") {
+      if (!dockerImageExists(docker_image)) {
+        dockerImageBuild(docker_image, ["pull": true])
+      }
+      else {
+        echo "Image ${docker_image} already exists."
+      }
+    }
+
+    stage("Test") {
+      parallel "Lint": {
+        dockerRun(docker_image, "flake8 snippets")
+      },
+      "Unit Test": {
+        def db_name = "mariadb-${env.GIT_COMMIT_SHORT}-${BUILD_NUMBER}"
+        def args = [
+          "docker_args": ("--name ${db_name} " +
+                          "-e MYSQL_ALLOW_EMPTY_PASSWORD=yes " +
+                          "-e MYSQL_DATABASE=snippets"),
+        ]
+
+        dockerRun("mariadb:10.0", args) {
+          args = [
+            "docker_args": ("--link ${db_name}:db " +
+                            "-e CHECK_PORT=3306 " +
+                            "-e CHECK_HOST=db")
+          ]
+          dockerRun("giorgos/takis", args)
```

Comment:
NIT: a comment explaining that takis waits for a port to open and quits might be helpful for future maintainers